dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/CacheHashMap.java

```diff
@@ -341,18 +394,18 @@ protected boolean persistSession(BackedSession d2, boolean propHit) {
     @Override
     protected BackedSession readFromExternal(String id) {
         String appName = getIStore().getId();
-        SessionKey key = new SessionKey(id, appName);
+        String key = createSessionKey(id, appName);
 
         CacheSession sess = null;
-        SessionData sessionData = cacheStoreService.cache.get(key);
+        ArrayList<?> sessionData = cacheStoreService.cache.get(key);
```

Comment:
It would still be useful to keep the `SessionData` class for the purpose of working with the field values in code, but instead of pushing the [de]serialization to the JCache layer, we could handle it in the session cache layer with some static helper method, such as:
```java
public static SessionData fromList(ArrayList<?> rawData);

public static ArrayList<?> toList(SessionData sessiondata);
```
This will be useful because we'll have type safety throughout the rest of our code, instead of needing to do things like `(Long) sessionData.get(LAST_ACCESS)`

Reply:
I suppose we could do that.  I'll work on making this change.